Update major version of Simulacra

daliwali opened the pull request:
This fixes a minor breaking change and includes a performance optimization specific for this benchmark: the preference of `appendChild` over `insertBefore`, which was the most significant slowdown.

krausest commented:
@daliwali 
I see two issues: Selecting a row has no effect and removing a row throws "Cannot read property 'splice' of undefined". Does it work for you?

daliwali commented:
@krausest sorry about that, I just fixed it and squashed the commit.

also it seems that there is still more optimization I have to do in order to make appending faster 🤔

krausest commented:
It's committed - though it turned out to be much harder than I expected. Nice speedup as can be seen [here](https://rawgit.com/krausest/js-framework-benchmark/8c6b10411089b2f28e174f495dc4ae9f193f2d03/webdriver-ts/table.html).

I changed two smaller things in your code:
- The remove icon is appended after the table to preload the icon, otherwise there are two repaints for the first time
- For the console logging I switched from requestAnimationFrame to window.setTimeout. The RAF is executed before the repaint and thus reports a too short duration.
<img width="883" alt="bildschirmfoto 2017-04-16 um 21 58 50" src="https://cloud.githubusercontent.com/assets/3374055/25074357/45854446-22f9-11e7-98f8-6ebfe6d122ce.png">
window.setTimeout is executed after repainting and is a better approximation:
<img width="1226" alt="bildschirmfoto 2017-04-16 um 22 00 00" src="https://cloud.githubusercontent.com/assets/3374055/25074364/66792c9e-22f9-11e7-81e4-8343f8dc3b08.png">

All of this wasn't problematic. It turned out that my test driver still had some issues which were revealed by your submission and thus helped to improve the test driver. The "create many rows" benchmark was reported as too fast (a negative duration...) and it took me quite some time nailing down that issue.
I had lots of tracing categories switched on and not all performance logging entries were returned on the driver.manage().logs() call. When I finally found that out I removed unneeded tracing categories. 
Sadly I had to find out that chromedriver 2.29.0 doesn't report any performance logs for selenium-webdriver 3.3.0 (though it seems to work with the old java testdriver) -I'll wait to see if an update to either fixes that issue....
